### PR TITLE
Support Search request slow logs and cluster.routing.allocation.awareness.force.zone.values

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description

- Coming from https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/logs/ and  support Search request slow logs.

- Coming from https://opensearch.org/docs/latest/tuning-your-cluster/ support `cluster.routing.allocation.awareness.force.zone.values` setting.

- Added support for the provider to onboard more cluster settings which are of type list `[]`, used `typeListClusterSettings`. 

### Issues Resolved
https://github.com/opensearch-project/terraform-provider-opensearch/issues/209 and https://github.com/opensearch-project/terraform-provider-opensearch/issues/202

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
### Testing

```
curl http://localhost:9200/_cluster/settings  -u 'admin:myStrongPassword123@456' | jq '.'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    32  100    32    0     0   9753      0 --:--:-- --:--:-- --:--:-- 10666
{
  "persistent": {},
  "transient": {}
}
```

With `terraform apply`
```
Terraform will perform the following actions:

  # opensearch_cluster_settings.persistent will be created
  + resource "opensearch_cluster_settings" "persistent" {
      + cluster_max_shards_per_node                    = 10
      + cluster_search_request_slowlog_level           = "WARN"
      + cluster_search_request_slowlog_threshold_debug = "2s"
      + cluster_search_request_slowlog_threshold_info  = "5s"
      + cluster_search_request_slowlog_threshold_trace = "100ms"
      + cluster_search_request_slowlog_threshold_warn  = "10s"
      + id                                             = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes
```
### Cluster output with applied Search request slow log settings


curl http://localhost:9200/_cluster/settings  -u 'admin:myStrongPassword123@456' | jq '.'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   265  100   265    0     0  74647      0 --:--:-- --:--:-- --:--:-- 88333
{
  "persistent": {
    "cluster": {
      "search": {
        "request": {
          "slowlog": {
            "level": "WARN",
            "threshold": {
              "warn": "10s",
              "trace": "100ms",
              "debug": "2s",
              "info": "5s"
            }
          }
        }
      },
      "max_shards_per_node": "10"
    },
    "plugins": {
      "index_state_management": {
        "template_migration": {
          "control": "-1"
        }
      }
    }
  },
  "transient": {}
}

```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -69,7 +69,7 @@ jobs:
         key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
         restore-keys: ${{ runner.os }}-go-
     - name: Run Docker containers
-      run: docker-compose up --detach
+      run: docker compose up --detach
```
Comment (peterzhuamazon) [inline .github/workflows/test.yml:72]: We sure to change to docker compose v2 now?
Will not break any existing functions?
If concern, switch back to v1 if needed.

File: docs/data-sources/host.md [hunk 2]
```diff
@@ -29,3 +29,5 @@ data "opensearch_host" "test" {
 
 - `id` (String) The ID of this resource.
 - `url` (String) the url of the active cluster
+
+
```
Comment (peterzhuamazon) [inline docs/data-sources/host.md:33]: Why added newlines?

File: docs/index.md [hunk 3]
```diff
@@ -87,6 +83,7 @@ EOF
 - `sniff` (Boolean) Set the node sniffing option for the OpenSearch client. Client won't work with sniffing if nodes are not routable.
 - `token` (String) A bearer token or ApiKey for an Authorization header, e.g. Active Directory API key.
 - `token_name` (String) The type of token, usually ApiKey or Bearer
+- `url` (String) OpenSearch URL
```
Comment (peterzhuamazon) [inline docs/index.md:86]: URL not Required anymore?

File: docs/resources/anomaly_detection.md [hunk 4]
```diff
@@ -78,3 +78,5 @@ EOF
 ### Read-Only
 
 - `id` (String) The ID of this resource.
+
+
```
Comment (peterzhuamazon) [inline docs/resources/anomaly_detection.md:82]: Same as above.

**Comment by prudhvigodithi** [comment]
Hey @peterzhuamazon the spaces coming from the CLI `tfplugindocs generate` https://github.com/hashicorp/terraform-plugin-docs which is used to generate the docs, example https://registry.terraform.io/providers/opensearch-project/opensearch/latest/docs/resources/cluster_settings.

File: .github/workflows/test.yml (hunk 1 above)
Comment (prudhvigodithi) [reply .github/workflows/test.yml:72]: Thanks, if the CI checks pass we should be good peter, this action is just used to spin up the OpenSearch cluster using docker compose and test the provider against that cluster.

File: docs/index.md (hunk 3 above)
Comment (prudhvigodithi) [reply docs/index.md]: I have updated this in my latest commit peter.

**Comment by prudhvigodithi** [comment]
To support the `cluster.routing.allocation.awareness.force.zone.values` with the list of string values example `:["zoneA", "zoneB"]`
```
curl http://localhost:9200/_cluster/settings  -u 'admin:myStrongPassword123@456' | jq '.'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   150  100   150    0     0  46210      0 --:--:-- --:--:-- --:--:-- 50000
{
  "persistent": {
    "cluster": {
      "routing": {
        "allocation": {
          "awareness": {
            "attributes": "zone",
            "force": {
              "zone": {
                "values": [
                  "zoneA",
                  "zoneB"
                ]
              }
            }
          }
        }
      }
    }
  },
  "transient": {}
}
```

**Review by peterzhuamazon — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks for the review @peterzhuamazon.
Adding @ironf1st @vinylen @getsaurabh02.

**Comment by vinylen** [comment]
Thank you for this!

**Pull request merged**